src/Workspaces/Core/Portable/Utilities/AsyncLazy`1.cs

```diff
@@ -294,6 +294,13 @@ private Request CreateNewRequest_NoLock()
 
         public override Task<T> GetValueAsync(CancellationToken cancellationToken)
```

Comment:
> My gut is saying do the check, the assumption being that checking the cancellation token is a handful of x86 instructions, and resuming a task on the thread pool only to immediately observe the cancellation token is thousands.

We want to prefer the fast path without sacrificing our overall goals with respect to cancellation. There is no need for cancellation checking at this fine of a granularity in order to preserve overall cancellation performance targets. One example of the same strategy can be seen in [Task.Wait](https://github.com/dotnet/coreclr/blob/e3b89a4abce3afb23ef92757cb77c7ade8d77862/src/mscorlib/src/System/Threading/Tasks/Task.cs#L2776-L2777).